### PR TITLE
feat: add docker image to install Argo CD extensions

### PR DESCRIPTION
**Pull request by leoluz**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM alpine:3.18.4
+
+ARG USER=ext-installer
+ENV HOME /home/$USER
+
+RUN apk update && apk add file curl
```
Comment (ashutosh16) [inline Dockerfile:6]: Does this command root access? you can downgrade the access later after installing.

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        mkdir -p /tmp/extensions/resources
+    fi
+    cp -Rf resources/* /tmp/extensions/resources/
+    echo "UI extension installed successfully"
+}
+
+
+## Script
+ext_enabled="${EXTENSION_ENABLED:-true}"
+if [ "$ext_enabled" != "true" ]; then
```
Comment (ashutosh16) [inline install.sh:72]:
```suggestion
if [[ "$ext_enabled" != "true" ]]; then
```

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    echo "Metrics extension is disabled"
+    exit 0
+fi
+
+ext_name="${EXTENSION_NAME:-}"
+ext_version="${EXTENSION_VERSION:-}"
+ext_url="${EXTENSION_URL:-}"
+if [ "$ext_url" == "" ]; then
```
Comment (ashutosh16) [inline install.sh:80]:
```suggestion
if [[ "$ext_url" == "" ]]; then
```

File: Dockerfile (hunk 1 above)
Comment (leoluz) [reply Dockerfile:6]: @ashutosh16 that is what I am doing at line 9

File: install.sh (hunk 3 above)
Comment (leoluz) [reply install.sh]: This is bash specific. Alpine/Busybox doesn't ship with bash. I want to have a minimalistic image to provide.

File: install.sh (hunk 2 above)
Comment (leoluz) [reply install.sh:72]: I checked and verified that Busybox [actually ships](https://www.busybox.net/downloads/BusyBox.html) with `[[` tests. However I am not using any specific feature from `[[` so I am keeping it standardized with the rest of the script. More on that [here](https://superuser.com/questions/1533900/difference-between-and-or-and-in-bash)

File: Dockerfile (hunk 1 above)
Comment (ashutosh16) [reply Dockerfile:6]: Got it. thx

File: install.sh (hunk 2 above)
Comment (ashutosh16) [reply install.sh:72]: Thank for the clarification.

**Review by ashutosh16 — APPROVED** [review]
LGTM!

**Pull request merged**